src/config.rs

```diff
@@ -333,32 +524,29 @@ impl Config {
     ) -> Result<T::Value> {
         config_map!(self, opts => ::internal::deserialize_from_custom_seed(seed, reader, opts))
     }
+}
 
-    /// Executes the acceptor with a serde::Deserializer instance.
-    /// NOT A PART OF THE STABLE PUBLIC API
-    #[doc(hidden)]
-    pub fn with_deserializer<'a, A, R>(&self, reader: R, acceptor: A) -> A::Output
-    where
-        A: DeserializerAcceptor<'a>,
-        R: BincodeRead<'a>,
-    {
-        config_map!(self, opts => {
-            let mut deserializer = ::de::Deserializer::new(reader, opts);
-            acceptor.accept(&mut deserializer)
-        })
+mod internal {
+    use super::*;
+    use byteorder::ByteOrder;
+
+    pub trait Options: Clone {
```

Comment:
I'm not 100% sure this `Clone` is needed. It is only passed to `ser::SizeChecker` to check a size limit, and to pre-allocate a Vec. It might be too much effort to remove this constraint though.